src/AccountDatabase.java

```diff
@@ -0,0 +1,318 @@
+            return; // Exit the method if the account is not found
+        }
+
+        Account accountInTheDatabase = accounts[accountToSearch];
+
+        // Call a deposit method on the account object with the specified depositAmount
+        accountInTheDatabase.makeDeposit(depositAmount);
+
+    }
+
+    /**
+     * Method to make a withdrawl
+     * @param account
+     * @return false if insufficient fund
+     */
+    public boolean withdraw(Account account, double withdrawalAmount) {
+        // Check if the account exists
+        if (!contains(account)) {
+            return false;
+        }
+
+        int accountToSearch = find(account);
+        Account accountInTheDatabase = accounts[accountToSearch];
+
+        // Check if there's enough balance for withdrawal
+        if (accountInTheDatabase.balance >= withdrawalAmount) {
+            // Specify the withdrawal amount here (dynamic)
+            accountInTheDatabase.makeWithdrawal(withdrawalAmount);
+            return true; // Withdrawal was successful
+        }
+
+        return false; // Withdrawal not allowed or insufficient balance
+    }
+
+
+    /**
+     * Method to print everything in the current Array
+     */
+    public void displayAllAccounts() {
+        for (int i = 0; i < numAcct; i++) {
+            Account account = accounts[i];
+            Profile profile = account.getProfileType();
+
+            String firstName = profile.getFirstName();
+            String lastName = profile.getLastName();
+            Date dateOfBirth = profile.getDateOfBirth();
+
+            String formattedInfo = firstName + " " + lastName + " " + dateOfBirth + " " + account.getAccountType();
+            System.out.println(formattedInfo);
+        }
+    }
+
+    @FunctionalInterface
+    private interface CustomComparator<T> {
+        /**
+         * Compares 2 things of type T
+         *
+         * @param a first thing to compare
+         * @param b second thing to compare
+         * @return negative number if a is less than b, positive number if a
+         * is greater than b, and 0 if they are equal
+         */
+        int compare(T a, T b);
+    }
+
+    /**
+     * Implementation of bubble sort. Could not do O(nlogn) sort because it
+     * had to be in place and don't have access to random library.
+     *
+     * @param toSort     array to be sorted
+     * @param numToSort  number of things to sort will only sort index 0
+     *                   through numToSort - 1. Any index after will be ignored
+     * @param comparator typically a lambda that will be used to compare two
+     *                   things in the array. Is expected to return a
+     *                   negative number if the first parameter is smaller
+     *                   than the second
+     * @param <T>        type of the array to be sorted
+     */
+    private static <T> void bubbleSort(T[] toSort, int numToSort, CustomComparator<T> comparator)
```

Comment:
I don't think your sort method needs to take a lambda. You only ever compare one way so you might as well call account.compareTo instead of doing it like this. Previous project required a comparator because we had to sort 3 different ways.